lxd/storage/drivers/driver_zfs.go

```diff
@@ -297,11 +297,18 @@ func (d *zfs) Delete(op *operations.Operation) error {
 	}
 
 	initialDatasets := d.initialDatasets()
+	deletedImgVolumePrefix := filepath.Join("deleted", BaseDirectories[VolumeTypeImage][0])
+
 	for _, dataset := range datasets {
 		if shared.StringInSlice(dataset, initialDatasets) {
 			continue
 		}
 
+		// Ignore deleted orphaned image volumes.
+		if strings.HasPrefix(dataset, deletedImgVolumePrefix) {
+			continue
+		}
```

Comment:
How is this happening exactly?

I purposefully didn't skip those as you shouldn't be allowed to delete a storage pool that has volumes left and when all instances are deleted, all deleted images should disappear as they'll be unused at that point.

A storage pool having leftover deleted image volumes indicates a bug in our instance deletion logic which should be fixed instead.

Reply:
I've removed this for now, it was occurring on my local system, but doesn't seem to be happening in the tests. Perhaps my local system had some other left over inconsistency I had mistakenly attributed to an issue with the recovery process (correctly) not restoring the image DB records.

Reply:
I've not been able to recreate so will leave out for now. Thats good.